plugins/woocommerce/src/RestApi/Routes/V4/ShippingZones/Method/Controller.php

```diff
@@ -0,0 +1,319 @@
+			}
+		} else {
+			// No zone_id provided, get the zone by method instance.
+			$zone = $this->validate_zone_by_method_instance( $instance_id );
+			if ( is_wp_error( $zone ) ) {
+				return $zone;
+			}
+		}
+
+		// Update method settings, enabled status, and order if any updates provided.
+		if ( isset( $request['enabled'] ) || isset( $request['settings'] ) || isset( $request['order'] ) ) {
+			$result = $method->update_from_api_request( $zone, $instance_id, $request->get_params() );
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		return rest_ensure_response( $this->method_schema->get_item_response( $method, $request ) );
+	}
+
+	/**
+	 * Get the schema for shipping methods.
+	 *
+	 * @return array
+	 */
+	protected function get_schema(): array {
+		return $this->method_schema->get_item_schema();
+	}
+
+	/**
+	 * Get error prefix for this controller.
+	 *
+	 * @return string
+	 */
+	protected function get_error_prefix(): string {
+		// Convert 'shipping-zone-method' to 'shipping_zone_method_'.
+		$prefix = str_replace( array( '-', '/' ), '_', $this->rest_base );
+		return 'woocommerce_rest_api_v4_' . $prefix . '_';
+	}
+
+	/**
+	 * Get the item response for a shipping method.
+	 *
+	 * @param mixed           $zone    Shipping method data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return array
+	 */
+	protected function get_item_response( $zone, WP_REST_Request $request ): array {
+		return $this->method_schema->get_item_response( $zone, $request, $this->get_fields_for_response( $request ) );
+	}
+
+	/**
+	 * Get route error by code, including custom shipping method errors.
+	 *
+	 * @param string $error_code Error code.
+	 * @return WP_Error
+	 */
+	protected function get_route_error_by_code( string $error_code ): WP_Error {
+		$custom_errors = array(
+			self::INVALID_ZONE_ID     => array(
+				'message' => __( 'Invalid shipping zone ID.', 'woocommerce' ),
+				'status'  => WP_Http::NOT_FOUND,
+			),
+			self::INVALID_METHOD_TYPE => array(
+				'message' => __( 'Invalid shipping method type.', 'woocommerce' ),
+				'status'  => WP_Http::BAD_REQUEST,
+			),
+			self::ZONE_MISMATCH       => array(
+				'message' => __( 'Shipping method does not belong to the specified zone.', 'woocommerce' ),
+				'status'  => WP_Http::BAD_REQUEST,
+			),
+		);
+
+		if ( isset( $custom_errors[ $error_code ] ) ) {
+			return $this->get_route_error_response(
+				$this->get_error_prefix() . $error_code,
+				$custom_errors[ $error_code ]['message'],
+				$custom_errors[ $error_code ]['status']
+			);
+		}
+
+		return parent::get_route_error_by_code( $error_code );
+	}
+
+	/**
+	 * Validate that a shipping zone exists.
+	 *
+	 * @param int $zone_id Zone ID.
+	 * @return WC_Shipping_Zone|WP_Error Zone object or error.
+	 */
+	protected function validate_zone( $zone_id ) {
+		$zone = WC_Shipping_Zones::get_zone( $zone_id );
+
+		if ( ! $zone || ( 0 !== $zone->get_id() && ! $zone->get_zone_name() ) ) {
```

Comment:
I'm not much into zone settings, but what's the reason for the second clause?

Reply:
The second clause `( 0 !== $zone->get_id() && ! $zone->get_zone_name() )` handles an edge case for corrupted or invalid zones. If a zone has an ID (is not zone 0 "Rest of World") AND has no name, then we throw error mentioning invalid.